### ISSUE TITLE
NBA Boxscrore Away, Home wins

### ISSUE THREAD

**Issue by Renaldas55** [roclark/sportsreference#230]
**Describe the bug**
Seems something wrong with NBA fields home_wins and away_wins. With newer data all is correct, but checking older counting seems incorrect.

**To Reproduce**
Just need Dataframe from 2018 and 2019 to run below 

data[data['Home_team']=='ORL'][['home_points','away_points','home_wins']]

**Screenshots**
Looking to below from fist season games, counter does not look good. 
![image](https://user-images.githubusercontent.com/56148524/66290309-340b0800-e8df-11e9-98b0-71210558f646.png)
Hera is example for newer data 2018, counter seems working well.
![image](https://user-images.githubusercontent.com/56148524/66290393-73d1ef80-e8df-11e9-81cc-966d0096713c.png)

**Desktop (please complete the following information):**
 - OS: [e.g. Ubuntu 18.04]
 - Sportsreference Version [e.g. 0.2.0 or commit hash if using code]

**Additional context**
Add any other context about the problem here. If applicable, also include links to any pages on www.sports-reference.com that are believed to be causing this issue.

**Comment by roclark** [comment]
Thanks for bringing this to my attention @Renaldas55! This is a puzzling issue, but it appears this is caused by the `Boxscore` class pulling the wrong record for the home team. It looks like the code is actually pulling the away team's record for the older games, which is why the numbers look off in the 2017-18 season. Luckily, I believe this is a pretty straightforward fix to get the right record. I will let you know when this is resolved. Thanks again!

**Comment by roclark** [comment]
@Renaldas55, I looked into this further, and I think I misinterpreted the question a bit. I found that the indexing was indeed incorrect in some scenarios (and have a fix coming), but I had to take another look at the definition of `home_wins`:

> Returns an `int` of the number of games the home team has won after the conclusion of the game.

While this is similar to a counter, it won't monotonically increase given it will show the season wins for the _home_ team after every game, and not just the team that is being queried. With this in mind, here's a breakdown of the beginning of Orlando's schedule in the 2017-18 season:
  * **Game 1:** Home Team: Magic, Home Team's Record: 1-0, `home_wins`: 1
  * **Game 2:** Home Team: Nets, Home Team's Record: 1-1, `home_wins`: 1
  * **Game 3:** Home Team: Cavaliers, Home Team's Record: 2-1, `home_wins`: 2
  * **Game 4:** Home Team: Magic, Home Team's Record: 3-1, `home_wins`: 3
  * **Game 5:** Home Team: Magic, Home Team's Record: 4-1, `home_wins`: 4
  * **Game 6:** Home Team: Hornets, Home Team's Record: 3-3, `home_wins`: 3

Notice that the number went down after game 6 in the Magic's schedule after they played the Hornets (with a worse record at the time) on the road. `home_wins` in this case refers to the Hornet's 3 games they've won during the season.

With all this being said, as I mentioned at the top, there's still an issue with the indexing that I need to fix in some scenarios and I will get something out shortly.

Thanks again for creating this issue and bringing it to my attention! Let me know if you have any other questions!
